portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,20 +13,101 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
 
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns data of comments*/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    public class Comment { 
```

Comment:
When you place a class inside of another class, the inner class will retain a reference to the instance of the outer class that it is constructed through. You can imagine that Java silently rewrites your class so it's like this:

```java
public class Comment {
    DataServlet outerClass;
    // ...
    public Comment(DataServlet outerClass, /* ... */) {
        this.outerClass = outerClass;
    }
}
```

This is sometimes useful, but often it's not what you intend to do. There are a couple ways you can avoid it. First is to just not nest the class; for something like comments, if  this wasn't a tiny demo project, it's likely it would be useful to create a separate complete comment class that handles some of the rules for manipulating stored comments independently of the servlet so that it can be re-used for different requests in different servlets.

In this case the simplest way to avoid it is to just use `public static class Comment`. `static class` in an inner-class declaration tells Java that you don't want the inner class to keep a reference to the outer class.

Reply:
done